website/server/models/user/methods.js

```diff
@@ -295,6 +296,13 @@ schema.statics.transformJSONUser = function transformJSONUser (jsonUser, addComp
   if (!jsonUser.flags.verifiedUsername) jsonUser.auth.local.username = null;
 
   if (addComputedStats) this.addComputedStatsToJSONObj(jsonUser.stats, jsonUser);
+
+  jsonUser.flags.newStuff = this.checkNewStuff(jsonUser.flags.lastNewStuffRead);
+};
+
+schema.statics.checkNewStuff = function checkNewStuff (lastReadID) {
```

Comment:
what's the difference between this static `checkNewStuff` and the method one defined below? Can we add a comment explaining the difference?

Reply:
Actually it looks like this one isn't used anywhere

Reply:
This one is used in the `transformJSONUser` method. Since that one is also static, I needed to add this static method because the method one isn't available. (If I misunderstood that somehow let me know. but it did give me an error before)

Reply:
You're right, but (this is super confusing and we should add at least a comment) `transformJSONUser` is only used [here](https://github.com/HabitRPG/habitica/blob/b947c714f0efb6de1b0558fff885913168890c82/website/server/controllers/api-v3/members.js#L398) because the User model isn't available when using `.lean()` and in that method `flags.newStuff ` should not be returned because it's not an user public field.

To get this field into normal user responses we should actually use the `schema.methods.checkNewStuff` in [here](https://github.com/HabitRPG/habitica/blob/b947c714f0efb6de1b0558fff885913168890c82/website/server/models/user/hooks.js#L25). Something like this

```
  toJSONTransform: function userToJSON (plainObj, originalDoc) {
    ...
    if (originalDoc.isSelected('flags.lastNewStuffRead')) { // make sure the necessary field to calculate newStuff are selected
        plainObj.flags.newStuff = originalDoc.checkNewStuff();
    }

    return plainObj;
  },
```